fe/src/main/java/org/apache/doris/load/loadv2/dpp/SparkRDDAggregator.java

```diff
@@ -0,0 +1,194 @@
+            // TODO: make more type aggregator
+            // replace ,sum,max,min
+            default:
+                throw new RuntimeException(String.format("unsupported aggregate type %s", aggType));
+        }
+    }
+
+}
+
+class EncodeMapFunction implements PairFunction<Row, Object[], Object[]> {
+
+    private SparkRDDAggregator[] valueAggregators;
+    // include bucket id
+    private int keyLen;
+
+    public EncodeMapFunction(SparkRDDAggregator[] valueAggregators, int keyLen) {
+        this.valueAggregators = valueAggregators;
+        this.keyLen = keyLen;
+    }
+
+    @Override
+    public Tuple2<Object[], Object[]> call(Row row) throws Exception {
+        Object[] keys = new Object[keyLen];
+        Object[] values = new Object[valueAggregators.length];
+
+        for (int i = 0; i < row.size(); i++) {
+            if (i < keyLen) {
+                keys[i] = row.get(i);
+            } else {
+                int valueIdx = i - keyLen;
+                values[valueIdx] = valueAggregators[valueIdx].buildValue(row.get(i));
+            }
+        }
+        return new Tuple2<>(keys, values);
+    }
+}
+
+class AggregateReduceFunction implements Function2<Object[], Object[], Object[]> {
+
+    private SparkRDDAggregator[] valueAggregators;
+
+    public AggregateReduceFunction(SparkRDDAggregator[] sparkDppAggregators) {
+        this.valueAggregators = sparkDppAggregators;
+    }
+
+    @Override
+    public Object[] call(Object[] v1, Object[] v2) throws Exception {
+        Object[] result = new Object[valueAggregators.length];
+        for (int i = 0; i < v1.length; i++) {
+            result[i] = valueAggregators[i].aggregate(v1[i], v2[i]);
+        }
+        return result;
+    }
+}
+
+
+
+class BitmapUnionAggregator extends SparkRDDAggregator<BitmapValue> {
+
+    @Override
+    BitmapValue aggregate(BitmapValue v1, BitmapValue v2) {
+        BitmapValue newBitmapValue = new BitmapValue();
+        if (v1 != null) {
+            newBitmapValue.or(v1);
+        }
+        if (v2 != null) {
+            newBitmapValue.or(v2);
+        }
+        return newBitmapValue;
+    }
+
+    @Override
+    byte[] serialize(Object value) {
+        try {
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            DataOutputStream outputStream = new DataOutputStream(bos);
+            ((BitmapValue)value).serialize(outputStream);
+            return bos.toByteArray();
+        } catch (IOException ioException) {
+            ioException.printStackTrace();
+            throw new RuntimeException(ioException);
+        }
+    }
+
+    @Override
+    BitmapValue buildValue(Object value) {
+        try {
+            BitmapValue bitmapValue = new BitmapValue();
+            if (value instanceof byte[]) {
+                bitmapValue.deserialize(new DataInputStream(new ByteArrayInputStream((byte[]) value)));
+            } else {
+                bitmapValue.add(value == null ? 0 : Long.valueOf(value.toString()));
```

Comment:
why value == null should add 0?

Reply:
global dict build won't deal null value ,so we need to deal it in dpp